pkg/controller/statusmanager/pod_status.go

```diff
@@ -247,13 +246,16 @@ func (status *StatusManager) setNodeConditionFromPod(podName types.NamespacedNam
 	// received the outdated pot event after the new pod running. For this case
 	// we should ignore the outdated event, otherwise, the operator may mistakenly
 	// set the NetworkUnavailbe status.
-	podStartedAt := pod.Status.StartTime.Time
-	nodeLastStartedAt := sharedInfo.LastNodeAgentStartTime[nodeName]
-	if podStartedAt.Before(nodeLastStartedAt) {
-		log.Info("Pod %s started at %v on node %s is outdated, there's new pod started at %v", podStartedAt, podName, nodeName, nodeLastStartedAt)
-		return
-	} else {
-		sharedInfo.LastNodeAgentStartTime[nodeName] = podStartedAt
+	// StartTime field will not be set until pod is initilized in pending state
+	if (pod.Status.StartTime != nil) {
+		podStartedAt := pod.Status.StartTime.Time
+		nodeLastStartedAt := sharedInfo.LastNodeAgentStartTime[nodeName]
+		if podStartedAt.Before(nodeLastStartedAt) {
+			log.Info(fmt.Sprintf("Pod %s started at %v on node %s is outdated, there's new pod started at %v", podStartedAt, podName, nodeName, nodeLastStartedAt))
```

Comment:
The log function here need to use fmt.Sprintf in order to include formatted print.